proto_parser/demo/demo_main.cc

```diff
@@ -0,0 +1,393 @@
+
+using ::wireless_android_play_analytics::ProtoValue;
+using ::wireless_android_play_analytics::MessageValue;
+using ::wireless_android_play_analytics::PrimitiveValue;
+using ::wireless_android_play_analytics::EventPredicate;
+using ::wireless_android_play_analytics::ProtoParser;
+
+namespace {
+
+class EventPredicateConverter {
+ public:
+
+  // Original:
+  //           # comment
+  //           # to
+  //           # save
+
+  //           # Condition 1
+  //           event_matcher {
+  //             input_field: "PlayExtension.store.click"
+  //           }
+  //           # Condition 2
+  //           event_matcher {
+  //             input_field: "PlayExtension.log_source"
+  //             int_comparator: 65  # WEB_STORE
+  //           }
+  //           # Condition 3
+  //           ui_element_path_predicate {
+  //             ui_element_type: STORE_UI_ELEMENT
+  //             # Condition 3.1
+  //             leaf_ui_predicate {
+  //               # ...
+  //               ui_element_type: STORE_UI_ELEMENT
+  //               ui_element_matcher {
+  //                 input_field: "type"
+  //                 int_comparator: 7305  # AUTHENTICATION_OPT_OUT_CHECKED
+  //               }
+  //             }
+  //             # Condition 3.2
+  //             index_ui_matcher {
+  //               ui_element_type: STORE_UI_ELEMENT
+  //               index: -2
+  //               ui_element_matcher {
+  //                 input_field: "type"
+  //                 int_comparator: 1  # SOMETHING
+  //               }
+  //             }
+  //           }
+  // Modified:
+  //           and {
+  //             # Condition 1
+  //             field_exists: "PlayExtension.store.click"
+  //             # Condition 2
+  //             field_value {
+  //               path: "PlayExtension.log_source"
+  //               equals_int: 65  # WEB_STORE
+  //             }
+  //             # Condition 3
+  //             ui_element {
+  //               type: STORE_UI_ELEMENT
+  //               and {
+  //                 # Condition 3.1
+  //                 leaf_ui {
+  //                   path:  "type"
+  //                   equals_int: 7305  # AUTHENTICATION_OPT_OUT_CHECKED
+  //                 }
+  //                 # Condition 3.2
+  //                 index_ui {
+  //                   index: -2
+  //                   path: "type"
+  //                   equals_int: 1  # SOMETHING
+  //                 }
+  //               }
+  //             }
+  //           }
+  std::unique_ptr<ProtoValue> ConvertEventPredicateFromFile(
+      absl::string_view path) {
+    ProtoParser parser(ReadTextProtoFromStream(path));
+    EventPredicate event_predicate_prototype;
+    std::unique_ptr<ProtoValue> event_predicate = 
+        parser.Create(event_predicate_prototype);
+    MessageValue* event_predicate_ptr = 
+        static_cast<MessageValue*>(event_predicate.get());  
+    // and_field is the master message that encloses everything.
```

Comment:
// We are converting event_matcher/ui_element_path_predicate to and here
// event_matcher {} => and {}
// ui_element_path_predicate{} => and{}